source/D2Common/src/Drlg/DrlgDrlg.cpp

```diff
@@ -545,47 +545,29 @@ int* __fastcall DRLG_GetRoomCenterX_RoomWarpXFromRoom(D2RoomExStrc* pRoomEx)
 //D2Common.0x6FD74B40
 void __fastcall DRLG_ComputeLevelWarpInfo(D2DrlgLevelStrc* pLevel)
 {
-	int* pX = NULL;
-	int* pY = NULL;
-	int nCounter = 0;
-	int nFlags = 0;
-	BOOL bHasWaypoint = FALSE;
-
 	for (D2RoomExStrc* pRoomEx = pLevel->pFirstRoomEx; pRoomEx; pRoomEx = pRoomEx->pRoomExNext)
 	{
-		if (pRoomEx->dwFlags & (ROOMEXFLAG_HAS_WAYPOINT | ROOMEXFLAG_HAS_WAYPOINT_SMALL))
-		{
-			bHasWaypoint = TRUE;
-		}
+		// First check if we have a waypoint
+		bool bHasWarp = (pRoomEx->dwFlags & ROOMEXFLAG_HAS_WAYPOINT_MASK);
 
-		if (pRoomEx->dwFlags & ROOMEXFLAG_HAS_WARP)
+		// Then check for additional warps
+		if (pRoomEx->dwFlags & ROOMEXFLAG_HAS_WARP_MASK && !bHasWarp)
 		{
-			if (!bHasWaypoint)
+			int nWarpIndex = 0;
+			for (int warpMask = ROOMEXFLAG_HAS_WARP_0; warpMask & ROOMEXFLAG_HAS_WARP_MASK; warpMask <<= 1)
 			{
-				nFlags = ROOMEXFLAG_HAS_WARP_0;
-				nCounter = 0;
-
-				while (!(nFlags & pRoomEx->dwFlags) || DRLGWARP_GetWarpDestinationFromArray(pLevel, nCounter) == -1)
-				{
-					nFlags <<= 1;
-					++nCounter;
-
-					if (!(nFlags & ROOMEXFLAG_HAS_WARP))
-					{
-						bHasWaypoint = FALSE;
-						break;
-					}
-				}
+				if (pRoomEx->dwFlags & warpMask && DRLGWARP_GetWarpDestinationFromArray(pLevel, nWarpIndex) != -1)
+					bHasWarp = true;
```

Comment:
It's definitely a matter of preference, but I usually tend to prefer braces after `if`, even for single line statements to make the scope more explicit